working/0723-static-extensions/feature-specification-variant1.md

```diff
@@ -0,0 +1,467 @@
+in order to manually resolve a name clash.*
+
+A static member invocation on a class `C`, of the form `C.m()` (or any
+other member access), is resolved by looking up static members in `C` named
+`m` and looking up static members of every accessible static extension with
+on-class `C` and a member named `m`.
+
+If `C` contains such a declaration then the expression is an invocation of
+that static member of `C`, with the same static analysis and dynamic
+semantics as before the introduction of this feature.
+
+Otherwise, an error occurs if no declarations named `m` or more than one
+declaration named `m` were found. *They would necessarily be declared in
+static extensions.*
+
+Otherwise, the invocation is resolved to the given static member
+declaration in a static extension named `Ej`, and the invocation is treated
+as `Ej.m()` *(this is an explicitly resolved invocation, which is specified
+above)*.
+
+#### The instantiated constructor return type of a static extension
+
+We associate a static extension declaration _D_ named `E` with formal type
+parameters `X1 extends B1 .. Xs extends Bs` and an actual type argument
+list `T1 .. Ts` with a type known as the _instantiated constructor return
+type of_ _D_ _with type arguments_ `T1 .. Ts`.
+
+When a static extension declaration _D_ named `E` has an on-clause which
+denotes a non-generic class `C`, the instantiated constructor return type
+is `C`, for any list of actual type arguments.
+
+*It is not very useful to declare a type parameter of a static extension
+which isn't used in the constructor return type, because it can only be
+passed in an explicitly resolved constructor invocation, e.g.,
+`E<int>.C(42)`. In all other invocations, the value of such type variables
+is determined by instantiation to bound. In any case, the type parameters
+are always ignored by static member declarations, they are only relevant to
+constructors.*
+
+When a static extension declaration _D_ has no formal type parameters, and
+it has an on-type `C<S1 .. Sk>`, the instantiated constructor return type
+of _D_ is `C<S1 .. Sk>`. *In this case the on-type is a fixed type (also
+known as a ground type), e.g., `List<int>`. This implies that the
+constructor return type of D is the same for every call site.*
+
+Finally we have the general case: Consider a static extension declaration
+_D_ named `E` with formal type parameters `X1 extends B1 .. Xs extends Bs`
+and a constructor return type `C<S1 .. Sk>`. With actual type arguments
+`T1 .. Ts`, the instantiated constructor return type of _D_ with type
+arguments `T1 .. Ts` is `[T1/X1 .. Ts/Xs]C<S1 .. Sk>`.
+
+#### Explicit invocation of a constructor in a static extension
+
+Explicit constructor invocations are similar to static member invocations,
+but they need more detailed rules because they can use the formal type
+parameters declared by a static extension.
+
+An _explicitly resolved invocation_ of a constructor named `C.name` in a
```

Comment:
I'm pretty lost in here.  Why do we support anything other than `E<S1 .. Ss>.C.name(args)`?  And what do the other forms mean?  Some examples or motivation might help here.